### ISSUE TITLE
How to construct the groundtruth file(pkl) if I want to test the code on my own dataset

### ISSUE THREAD

**Issue by qcl1994** [filipradenovic/cnnimageretrieval-pytorch#3]
Hi,Filip
Thank you for your open-source code! Could you tell me How to construct the groundtruth file(pkl) if I want to test the code on my own dataset?

**Comment by filipradenovic** [comment]
Hello, take a look at [testdataset.py](https://github.com/filipradenovic/cnnimageretrieval-pytorch/blob/master/cirtorch/datasets/testdataset.py) script. The .pkl gnd file should be a dictionary containing:

- 'imlist': list of database images
- 'qimlist': list of query images
- 'gnd': list with the same length as qimlist, each element is a dictionary containing: 'bbx' - query bounding box; 'ok' or 'easy' or 'hard' or 'junk' - index list of positive or easy or hard or junk images.

Then, it should be easy to add a new dataset in the `testdataset.py`.

**Comment by ivicts** [comment]
Hi if let's say if I wanted to train on oxford5k dataset and test the code on oxford5k dataset (without finetuning), can I do this?
python3 -m cirtorch.examples.train YOUR_EXPORT_DIR --gpu-id '0' --training-dataset 'oxford5k' 
            --test-datasets 'oxford5k,paris6k' --arch 'resnet101' --pool 'gem' --loss 'contrastive' 
            --loss-margin 0.85 --optimizer 'adam' --lr 1e-6 --neg-num 5 --query-size=2000 
            --pool-size=20000 --batch-size 5 --image-size 362

Does your code restricted only to fine tune from the Structure-from-Motion (SfM) method or it can also train from oxford5k from scratch and test on oxford5k? I wanted to adapt the dataset to problem other than landmark or building retrieval.
how about the performance comparison with other state of the art without fine tuning and your network without fine tuning?

**Comment by filipradenovic** [comment]
For now, in this codebase, only retrieval-SfM-120k is available for training. However, any dataset that can be saved/loaded in the db format that training requires, can be used for training. You can create a database from any dataset you want, and easily change the code to train on it. It should be simple enough to add new testing datasets as well.

It doesn't make sense to train on oxford5k and evaluate on it, as you would use ground truth for training, while it should be used for evaluation exclusively. 

To train the network from "scratch", ie, not initializing with ImageNet, you can run the training script with additional flag `--not-pretrained`. 

Finally, test script supports evaluation on our fine-tuned networks, off-the-shelf networks pre-trained on ImageNet, and any network you trained (fine-tuned or from scratch), see [README.md](https://github.com/filipradenovic/cnnimageretrieval-pytorch/blob/master/README.md) for more details.
